toolkit/tools/imagegen/configuration/configuration.go

```diff
@@ -122,6 +122,29 @@ func checkDeviceMapperFlags(config *Config) (err error) {
 	return
 }
 
+// checkInvalidMountIdentifiers checks that we don't have an invalid combination of GPT/MBR, PARTLABEL, and Name for each partition.
+// PARTUUID and PARTLABEL are GPT concepts. MBR partly supports PARTUUID, but is completely incompatible with PARTLABEL.
+// If we want to use PARTLABEL, we need to define a [Name] for the partition as well.
+func checkInvalidMountIdentifiers(config *Config) (err error) {
+	for _, sysConfig := range config.SystemConfigs {
+		for _, partSetting := range sysConfig.PartitionSettings {
+			if partSetting.MountIdentifier == MountIdentifierPartLabel {
+				diskPart := config.GetDiskPartByID(partSetting.ID)
+				disk := config.GetDiskContainingPartition(diskPart)
+
+				if disk.PartitionTableType != PartitionTableTypeGpt {
+					return fmt.Errorf("[SystemConfig] '%s' mounts a [Partition] '%s' via PARTLABEL, but that partition is on an MBR disk which does not support PARTLABEL", sysConfig.Name, partSetting.ID)
+				}
+
+				if diskPart.Name == "" {
```

Comment:
Assuming there's a name length limit when using PARTLABEL this would be a good place to validate and error before the underlying partition tools do it for us.

Reply:
Added the check to the `Partition` object itself. We only care about a minimum length if we are using the PARTLABEL mode, but we always need to check for name overflow.